packages/hierarchy-builder/src/hierarchy-builder/Utils.ts

```diff
@@ -3,8 +3,31 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+import { ECClass, IMetadataProvider } from "./ECMetadata";
+import { getClass } from "./internal/Common";
+
 /**
  * This is an utility `Omit` type which works with union types.
  * @beta
  */
 export type OmitOverUnion<T, K extends PropertyKey> = T extends T ? Omit<T, K> : never;
+
+/** @beta */
+export class BaseClassChecker {
+  private _map = new Map<string, boolean>();
+
+  public async isECClassOfBaseECClass(metadata: IMetadataProvider, ecClassNameToCheck: string, baseECClass: ECClass): Promise<boolean> {
```

Comment:
any reason not to make `metadata` a class variable?

Reply:
Changed.